cpp/src/gandiva/precompiled/time.cc

```diff
@@ -241,6 +241,44 @@ int getJanWeekOfYear(const EpochTimePoint& tp) {
   return 52;
 }
 
+static const char* WEEK[] = {"SUNDAY",   "MONDAY", "TUESDAY", "WEDNESDAY",
+                             "THURSDAY", "FRIDAY", "SATURDAY"};
+
+static const int WEEK_LEN[] = {6, 6, 7, 9, 8, 6, 8};
+
+#define NEXT_DAY_FUNC(TYPE)                                                             \
+  FORCE_INLINE                                                                          \
+  gdv_date64 next_day_from_##TYPE(gdv_int64 context, gdv_date64 millis, const char* in, \
```

Comment:
I think you should replace `gdv_date64 millis` to `gdv_##TYPE millis` to ensure it can receive all possible parameters.